controls/migrations/0055_auto_20210726_2246.py

```diff
@@ -0,0 +1,33 @@
+# Generated by Django 3.2.5 on 2021-07-26 22:46
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('controls', '0054_merge_0052_auto_20210521_1422_0053_auto_20210701_1133'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='historicalstatement',
+            name='source',
+            field=models.CharField(blank=True, help_text="Statement source such as '../../../nist.gov/SP800-53/rev4/json/NIST_SP-800-53_rev4_catalog.json'.", max_length=200, null=True),
+        ),
+        migrations.AddField(
+            model_name='statement',
+            name='source',
+            field=models.CharField(blank=True, help_text="Statement source such as '../../../nist.gov/SP800-53/rev4/json/NIST_SP-800-53_rev4_catalog.json'.", max_length=200, null=True),
+        ),
+        migrations.AlterField(
+            model_name='historicalstatement',
+            name='statement_type',
+            field=models.CharField(blank=True, choices=[('CONTROL_IMPLEMENTATION', 'control_implementation'), ('CONTROL_IMPLEMENTATION_LEGACY', 'control_implementation_legacy'), ('CONTROL_IMPLEMENTATION_PROTOTYPE', 'control_implementation_prototype'), ('ASSESSMENT_RESULT', 'assessment_result'), ('POAM', 'POAM'), ('SECURITY_SENSITIVITY_LEVEL', 'security_sensitivity_level'), ('SECURITY_IMPACT_LEVEL', 'security_impact_level')], help_text='Statement type.', max_length=150, null=True),
+        ),
+        migrations.AlterField(
+            model_name='statement',
+            name='statement_type',
+            field=models.CharField(blank=True, choices=[('CONTROL_IMPLEMENTATION', 'control_implementation'), ('CONTROL_IMPLEMENTATION_LEGACY', 'control_implementation_legacy'), ('CONTROL_IMPLEMENTATION_PROTOTYPE', 'control_implementation_prototype'), ('ASSESSMENT_RESULT', 'assessment_result'), ('POAM', 'POAM'), ('SECURITY_SENSITIVITY_LEVEL', 'security_sensitivity_level'), ('SECURITY_IMPACT_LEVEL', 'security_impact_level')], help_text='Statement type.', max_length=150, null=True),
```

Comment:
I don't understand the need for the change `statement_type` field. We are using `.name` value with our `StatementTypeEnum` throughout our code such as line 536 in `controls/models.py`:

```
        self.root_element.statements_consumed.filter(statement_type=StatementTypeEnum.CONTROL_IMPLEMENTATION.name,
                                                     sid_class=control.oscal_catalog_key,
                                                     sid=control.oscal_ctl_id
                                                     ).delete()
```

It seems like this migration specifies lowercase. 

Here is an example control from my current database. The value `statement_type` is in upper case rather than lower case.

```
In [1]: smt = Statement.objects.all()[100]

In [2]: smt
Out[2]: 'CONTROL_IMPLEMENTATION ac-14 b NIST_SP-800-53_rev4 1392'

In [3]: smt.__dict__
Out[3]: 
{'_state': <django.db.models.base.ModelState at 0x7fdd7a60e2b0>,
 'id': 1392,
 'sid': 'ac-14',
 'sid_class': 'NIST_SP-800-53_rev4',
 'pid': 'b',
 'body': 'This control is reviewed at least annually or as needed by the ISSO and SO.\n2 Twelve Responsibility:\nNo actions are allowed in the environment without authentication. The requirements for identification and authentication are documented within this SSP. For further details of the identification and authentication requirements, review IA-2 for user identification and IA-3 for device authentication.\nApplication, Database, Operating System:  No actions are allowed in the environment without authentication. The requirements for identification and authentication to the application, and operating system are documented within this SSP. Customer applications authentication requires MFA or SSO through SAML 2.0. All E3LabIaas access to 2 Twelve admins to perform administrative activities is provided through SSH connections to the host.\nCustomer Responsibility:\nIt is the customer responsibility to ensure that all actions that can be performed without identification or authentication are fully documented.',
 'statement_type': 'CONTROL_IMPLEMENTATION',
 'remarks': None,
 'status': 'Implemented',
 'version': None,
 'created': datetime.datetime(2020, 9, 4, 9, 40, 36, 937162, tzinfo=<UTC>),
 'updated': datetime.datetime(2020, 12, 7, 19, 36, 33, 434548, tzinfo=<UTC>),
 'parent_id': None,
 'prototype_id': 749,
 'producer_element_id': 11,
 'consumer_element_id': 26,
 'uuid': UUID('bb81bbe7-b22e-4b87-bc24-6257b12585eb'),
 'import_record_id': None,
 '_peers': <WeakValueDictionary at 0x7fdd78a99610>}
```

Reply:
I dont see anything in the accompanying [ commit](https://github.com/GovReady/govready-q/commit/a95d38c6fa9a39683228818fed0d89e2d6742f57) that actually touches statement_type. This commit handles adding source to statements that might be why it is showing up as historical statement needs all of statement's fields to be recorded. Sound about right to you @gregelin ?